session/general_log.go

```diff
@@ -102,11 +105,21 @@ func newGeneralLogger(logger *zap.Logger) *generalLogger {
 	return gl
 }
 
+func userToString(user auth.UserIdentity) string {
+	buf := stringBufferPool.Get()
+	buf.WriteString(user.Username)
+	buf.WriteByte('@')
+	buf.WriteString(user.Hostname)
+	ret := buf.String()
+	buf.Free()
+	return ret
```

Comment:
Is it possible to implement this at `UserIdentity`? For example,

```go
func (i UserIdentity) ToString() string {
  ...
}
```

Reply:
> Looks like the dependency problem you met is mainly caused by putting the business logic in the wrong place. From the name, `logutil` is a utility package, which should be **imported by most of the other packages**. However it is obvious that "general log" is not something that can be imported by other packages. Instead, "general log" should import others, in order to correctly indicate what kind of fields it needs. This clearly indicates that the current package arrangement is wrong. This can be discovered by the name as well, as something should be either general (utility), or business-related. Now logutil is called a util, while general-log is a business feature.
> 
> I'm thinking that the package could be `session/generallog/...`. Notice that as it is an independent package now, the parent directory doesn't really matter.

Yes, and I've already moved it into `session/`, and trying to remove all the `genXxx` functions.

Reply:
Yes, and since it's in `pingcap/parser`, I'll leave a todo here for now.